Merge remote-tracking branch 'origin/alaska-prod' into alaska-alt-1

jovial opened the pull request.

JohnGarbutt committed:
Enable Manila with CephFS native backend

Uses upstream support in kolla ansible for Manila with the CephFS backend.

jovial committed:
Merge commit '302babfe14685104770dfb61ce03bcc276a805f7' into sync-take-2

jovial committed:
Revert "[UKT0] Add support for Keystone authentication via EGI's Dev AAI"

This reverts commit 3dd18c34d4f61a2c7bec1d361ad0d220dc10bd41.

markgoddard committed:
Merge pull request #88 from jovial/changes-from-alt1

Synchronize changes from alt-1

jovial committed:
Merge remote-tracking branch 'kayobe-config/stable/pike' into merge/kayobe-config-upstream-prod

markgoddard committed:
Merge pull request #91 from jovial/merge/kayobe-config-upstream-prod

Merge openstack/kayobe-config#stable/pike into prod

jovial committed:
Merge tag 'alaska-alt-1-3.1.0.1' into sync/prod-pike-latest

Differences:

These files are related to EGI checkin support, which has not yet been
enabled on production:

* etc/kayobe/kolla/config/horizon/local_settings
* etc/kayobe/kolla/config/keystone/wsgi-keystone.conf

They were therefore not merged and only present on alt-1.

- etc/kayobe/networks.yml contains address pool allocation changes which
  should not be merged
- etc/kayobe/kolla/globals.yml set `keystone_tag: 6.0.0.0` which is related
  to EGI checkin work
- opensm is disabled on alt-1: the production and alt-1 partitions share an
  infiniband network and there can only be one instance of subnet manager at a time

jovial committed:
Merge remote-tracking branch 'origin/alaska-alt-1' into sync/queens

Brings in the changes necessary to update to queens.

Differences from alt-1:

- EGI checkin support not enabled

markgoddard committed:
Merge pull request #102 from jovial/sync/prod-pike-latest

Sync/prod pike latest

markgoddard committed:
Use the same base_path variable for production and alt-1

This affects where virtualenvs, images, source code, and other things are
created.

Commit:
Merge pull request #105 from SKA-ScienceDataProcessor/fix-base-path

Use the same base_path variable for production and alt-1

markgoddard committed:
Re-enable monasca

We are using tags to skip monasca in the upgrade.

markgoddard committed:
Merge pull request #103 from jovial/sync/queens

Sync alt-1 queens changes

markgoddard committed:
Migrate Kolla Ansible custom inventory to Queens

markgoddard committed:
Merge pull request #106 from SKA-ScienceDataProcessor/migrate-kolla-inventory

Migrate Kolla Ansible custom inventory to Queens

markgoddard committed:
Merge branch 'manila-queens-alt-1' into manila-queens-prod

markgoddard committed:
Merge branch 'update-magnum-alt-1' into update-magnum-prod

Conflicts:
	etc/kayobe/kolla/globals.yml

markgoddard committed:
Merge pull request #107 from SKA-ScienceDataProcessor/manila-queens-prod

Manila queens prod

Commit:
Merge pull request #110 from SKA-ScienceDataProcessor/update-magnum-prod

Update magnum (prod)

dougszumski committed:
Upgrade Monasca to Queens

* Upgrade InfluxDB to latest version in Queens devstack. We could
  probably bump this even further before we hit breaking changes.
* Use the InfluxDB config file from K-A - we don't need to
  use a pre-historic version of it anymore.
* Fluentd Monasca plugin is now baked into the Fluentd image from the
  factory
* Remove redundant setting of enable flags
* Use latest Kafka from K-A queens and set the messaging version to the
  latest version supported by Monasca
* Use latest Zookeeper from K-A queens.
* Add Storm service

dougszumski committed:
Merge pull request #112 from SKA-ScienceDataProcessor/feature/monasca_queens

Upgrade Monasca to Queens

dougszumski committed:
Disable max series and values per tag settings in InfluxDB

Although it would be better to fix the cause, this change allows InfluxDB
to continue accepting new metrics until a better solution is cobbled
together. These limits don't appear to have been present in 0.10 from
which we recently upgraded.

dougszumski committed:
Merge pull request #113 from SKA-ScienceDataProcessor/relax_influx_limits

Disable max series and values per tag settings in InfluxDB

dougszumski committed:
Disable InfluxDB reporting

dougszumski committed:
Enable Monasca Notification Slack plugin

* Install bug fixes to make plugin work
* Provide a template for the Slack notifications

dougszumski committed:
Add release tags for Monasca

dougszumski committed:
Merge pull request #115 from SKA-ScienceDataProcessor/monasca_notif_update

Monasca notif update

dougszumski committed:
Merge branch 'alaska-alt-1' into alaska-prod

Conflicts:
	etc/kayobe/kolla.yml

dougszumski committed:
Merge pull request #117 from SKA-ScienceDataProcessor/bugfix/docker_logs_prod

Bugfix/docker logs prod

dougszumski committed:
Update Monasca config

Update Monasca persister tag to include bugfix
Add custom log metrics config file for creating metrics from logs

dougszumski committed:
Merge pull request #118 from SKA-ScienceDataProcessor/monasca_conf_update

Update Monasca config

markgoddard committed:
Merge branch 'alaska-alt-1' into sync-alt-1

markgoddard committed:
Merge pull request #124 from SKA-ScienceDataProcessor/sync-alt-1

Sync alt-1 into production

jovial committed:
Add serial console allocation file

The serial consoles have been enabled with:

  kayobe baremetal compute serial console enable

These are the resulting TCP port allocations.

Commit:
Merge branch 'alaska-alt-1' into alaska-prod

Conflicts:
	etc/kayobe/inventory/host_vars/ethsw-b16-u38
	etc/kayobe/inventory/host_vars/ethsw-b17-u42
	etc/kayobe/kolla/globals.yml

markgoddard committed:
Merge pull request #129 from brtknr/alaska-prod

Merge alt-1 into production

markgoddard committed:
Merge pull request #125 from jovial/consoles

Add serial console allocation file

dougszumski committed:
Avoid reaping Kafka connections

Kafka introduced an optimisation in 0.10+ that reaps idle socket
connections. Kafka clients were generally updated to support
closing connections before they were reaped / re-opening new ones,
however - *please sit down* - Monasca forked the kafka-python library
just before this support was added. This results in a slurry of errors
relating to closed connections from the various Monasca services using
the forked client. The simplest way to fix this problem is to turn off
the connection reaping on the broker side by setting the timeout to
an arbitrarily long time, in this case 10 years (field is a Java long).
It is difficult to backport the patch for connection reaping to the
kafka-python fork because asynchronous methods were introduced after
the Monasca fork, and are not used by Monasca. When Monasca moves to
a modern Kafka library this patch can be reverted.

dougszumski committed:
Merge pull request #134 from SKA-ScienceDataProcessor/bugfix/avoid_reaping_kafka_connections

Avoid reaping Kafka connections

dougszumski committed:
Remove legacy LXC logical volume

LXC Monasca services have been replaced with Kolla managed Docker
services.

dougszumski committed:
Merge pull request #135 from SKA-ScienceDataProcessor/remove_lxc_volume

Remove legacy LXC logical volume

dougszumski committed:
Upgrade Monasca to Rocky release

This adds the following new password to passwords.yml:

- monasca_grafana_admin_password

This also moves the following password from secrets.yml
to passwords.yml as it is now managed by Kolla-Ansible:

- monasca_agent_password

markgoddard committed:
Merge tag 'alaska-alt-1-4.1.0' into alaska-prod

markgoddard committed:
Merge tag 'alaska-alt-1-5.0.0' into alaska-prod

markgoddard committed:
Update passwords for Rocky

dougszumski committed:
Changes to reflect current state of Monasca deployment

dougszumski committed:
Support using legacy Grafana until bug is fixed

markgoddard committed:
Merge pull request #139 from SKA-ScienceDataProcessor/update-queens-prod

Update queens prod

dougszumski committed:
Upgrade InfluxDB to latest release

InfluxDB 1.3 contains a bug which allows a tenant to view
dimensions from all OpenStack projects. This is a security
issue.

markgoddard committed:
Merge pull request #137 from SKA-ScienceDataProcessor/feature/prod_monasca_rocky

Upgrade to Rocky release

dougszumski committed:
Add vanilla influxdb config file

dougszumski committed:
Enable the TSI disk based index

Using a disk based index allows high cardinality datasets to
be used without burning through 200+GB of memory.

Note that you need to follow the migration procedure to
convert the index.

See doc:

https://docs.influxdata.com/influxdb/v1.7/administration/config/#index-version-inmem

Migration guide:

https://docs.influxdata.com/influxdb/v1.7/administration/upgrading/#upgrading-influxdb-1-3-1-4-no-tsi-preview-to-1-7-x-tsi-enabled

dougszumski committed:
Disable InfluxDB series limits

Using the disk based time series index allows us to relax these
limits.

dougszumski committed:
Bump Kolla tags to pull in bug fixes

This pulls in fixes for Fluentd config, the Monasca
log query API and the Monasca Notification service.

dougszumski committed:
Disable storing alarm history from Kafka in InfluxDB

Until the below is fixed this feature is broken so we
should turn it off to avoid spamming the logs with
error messages.

https://storyboard.openstack.org/#!/story/2005434

dougszumski committed:
Use old Grafana image until bug fixed in Rocky build

Although the Grafana code hasn't changed in Rocky, the dependencies
have and this has caused some graphical errors. We need to find out
what has changed and pin it. Until then, use the old container.

dougszumski committed:
Merge pull request #141 from SKA-ScienceDataProcessor/monasca_upgrade

Monasca Rocky upgrade: Bugfixes and InfluxDB

dougszumski committed:
Move to Monasca Grafana

In Rocky a dedicated Monasca Grafana container exists. Here
we complete the switch to it.

This also pulls in a fix for Grafana org registration.

dougszumski committed:
Merge pull request #143 from SKA-ScienceDataProcessor/monasca_grafana

Move to Monasca Grafana

dougszumski committed:
Merge branch 'alaska-alt-1' into alaska-prod

Conflicts:
	etc/kayobe/kolla.yml

dougszumski committed:
Merge branch 'alaska-alt-1' into token_branch_bump

dougszumski committed:
Update override to point to production controller

dougszumski committed:
Merge pull request #146 from SKA-ScienceDataProcessor/token_rotate_prod

Bump KA version to include token rotation fix

dougszumski committed:
Merge branch 'alaska-alt-1' into alaska-prod

dougszumski committed:
Merge pull request #151 from SKA-ScienceDataProcessor/grafana_roles_prod

Grafana roles prod

Commit:
Merge branch 'alaska-alt-1' into alaska-prod

Commit:
Merge pull request #153 from brtknr/alaska-prod

 Bump up kolla-ansible version and whitelist rabbitmq queues

markgoddard committed:
Merge branch 'alaska-alt-1' into alaska-prod

Commit:
Merge pull request #155 from brtknr/alaska-prod

Make zsh compatible

Commit:
Merge branch 'alaska-alt-1' into alaska-prod

Commit:
Merge pull request #157 from brtknr/alaska-prod

 Support time range in dimension name/value queries (prod)

dougszumski committed:
Merge branch 'alaska-alt-1' into alaska-prod

dougszumski committed:
Merge pull request #159 from SKA-ScienceDataProcessor/add_isaac_user

Add isaac user

dougszumski committed:
Merge pull request #162 from SKA-ScienceDataProcessor/alaska-alt-1

Alaska alt 1 merge

brtkwr committed:
Merge branch 'alaska-alt-1' into alaska-prod

Conflicts:
	etc/kayobe/kolla/config/manila/manila-share.conf

dougszumski committed:
Merge pull request #165 from brtknr/alaska-prod

Manila config

Wasaac committed:
Merge branch 'alaska-alt-1' into alaska-prod

dougszumski committed:
Merge pull request #167 from Wasaac/alaska-prod

Merge alt-1 monasca thresh fixes

Wasaac committed:
Merge branch 'alaska-alt-1' into alaska-prod

dougszumski committed:
Merge pull request #169 from Wasaac/alaska-prod-merge

Alaska prod merge

Wasaac committed:
Merge branch 'alaska-alt-1-monasca-grafana' into alaska-prod-monasca-grafana

dougszumski committed:
Merge pull request #173 from Wasaac/alaska-prod-monasca-grafana

Alaska prod monasca grafana

brtkwr committed:
Support read and write to database per tenancy (#175)

Wasaac committed:
Revert "Support read and write to database per tenancy (#175)" (#177)

This reverts commit 780fc60e0bf52e4b54efd3a3a5fb8bb8314aa8c4.

Wasaac committed:
Merge monasca alarms and multi-tenant influxdb changes from alt-1 (#181)

* Support read and write to database per tenancy (#178)

* Monasca Alarms playbook (#174)

* Use Slack notifications template feature

Adds a slack webhook notification template linking alarm
notifications to a grafana.

* Add custom playbook to create monasca alarms

Uses stackhpc.monasca_default_alarms ansible galaxy role to
configure default monasca alarms and notification method.
Also adds the necessary symlinks to support custom playbooks.

markgoddard committed:
Merge alaska-alt-1 into alaska-prod

markgoddard committed:
Merge pull request #184 from SKA-ScienceDataProcessor/update-neutron-prod

Sync production with alt-1

jovial committed:
Merge remote-tracking branch 'origin/alaska-prod' into rocky-alt-1

jovial commented:
Dammit - ended up committing changes I didn't want.

jovial commented:
Turns out that I had selected the wrong base for the PR. Ready to review.

jovial commented on etc/kayobe/opensm.yml:
```diff
@@ -4,7 +4,7 @@
 # Whether OpenSM is enabled.
 #opensm_enabled:
-opensm_enabled: False
+opensm_enabled: True
```
Is the subnet manager shared across production and alt-1?

markgoddard replied:
Yes - just one IB network

markgoddard commented:
Why are we merging prod into alt-1? Normally we do the reverse.

markgoddard commented on etc/kayobe/kolla/inventory/overcloud-components.j2:
```diff
@@ -0,0 +1,280 @@
+# This inventory section provides a mapping of the components to top level
```
Sure we want this? alt-1 monitoring is on the controller, prod monitoring is on a separate host.

jovial replied:
we define:
```
kolla_overcloud_inventory_top_level_group_map:
  control:
    groups:
      - controllers
  network:
    groups:
      - network
  compute:
    groups:
      - compute
  monitoring:
    groups:
      - controllers
```
https://github.com/SKA-ScienceDataProcessor/alaska-kayobe-config/blob/29429c5264eb0d6eed20b668231bc7ac357d23b3/etc/kayobe/kolla.yml#L214

So it should be equivalent, right? My opinion is that the more similar the config is the better.

jovial commented:
> Why are we merging prod into alt-1? Normally we do the reverse.

So that we test an upgrade with equivalent config. Some stuff has been committed to production that is not in alt-1.

jovial committed:
Disable opensm again

This corrects a mistake in the merge resolution. There is a single
shared IB network between production and alt1.
